Source/Application/Kysect.Shreks.Application.Handlers/Google/GetCoursePointsBySubjectCourseHandler.cs

```diff
@@ -51,22 +51,30 @@ private StudentPointsDto GetStudentPoints(Student student, IQueryable<Submission
             .Where(s => s.Student.Equals(student))
             .AsEnumerable()
             .GroupBy(s => s.Assignment)
-            .Select(s => GetAssignmentPoints(s.ToArray()))
+            .Select(s => FindAssignmentPoints(s.ToArray()))
+            .Where(a => a is not null)
+            .Select(a => a!)
             .ToArray();
 
         var studentDto = _mapper.Map<StudentDto>(student);
 
         return new StudentPointsDto(studentDto, assignmentPoints);
     }
 
-    private AssignmentPointsDto GetAssignmentPoints(IEnumerable<Submission> submissions)
+    private AssignmentPointsDto? FindAssignmentPoints(IEnumerable<Submission> submissions)
     {
-        var submission = submissions.OrderBy(s => s.SubmissionDate).First();
+        Submission submission = submissions
+            .Where(s => s.Points != null)
+            .OrderBy(s => s.SubmissionDate)
+            .First();
+
         //TODO: add deadlines usage instead of .Last
 
         var points = submission.Points;
+        if (points is null)
+            return null;
 
         var submissionDate = submission.SubmissionDate;
-        return new AssignmentPointsDto(submission.Assignment.Id, submissionDate, points?.Value);
+        return new AssignmentPointsDto(submission.Assignment.Id, submissionDate, points.Value.Value);
```

Comment:
```suggestion
        Submission? submission = submissions
            .Where(s => s.Points is not null)
            .MinBy(s => s.SubmissionDate);

        //TODO: add deadlines usage instead of .Last

        if (submission is null)
            return null;

        var submissionDate = submission.SubmissionDate;
        var submissionPoints = submission.Points!.Value;

        return new AssignmentPointsDto(submission.Assignment.Id, submissionDate, submissionPoints.Value);
```